spec/features/welcome_spec.rb

```diff
@@ -2,7 +2,24 @@
 
 feature 'visiting the homepage' do
   scenario 'the visitor sees the search title' do
-    visit root_path
+    visit '/'
     expect(page).to have_text("Search")
   end
+
+  scenario 'the visitor can input text and search result number and search' do
+    visit '/'
+    fill_in('search_term_input', :with => 'water')
+    fill_in('result_count', :with => '50')
+    click_button 'Search'
+    expect(page).to have_text("water")
+    expect(page).to have_css("div.changeless_box", :between => 50..100)
+  end
+
+  scenario 'when "search" is clicked without search term top 10 results are displayed' do
+    visit '/'
+    click_button 'Search'
+    expect(page).to have_css("div.changeless_box", :between => 10..20)
```

Comment:
Does this only pass if there are no changes?

I would avoid relying too much on what search actually returns in these tests, because your tests should always give the same result if the code is the same.

In this case we're lucky that the dependency (the search API) is public, pretty reliable, and calling it has no side effects. But I just wanted to point out that in general, with integration testing you usually have to do some special setup/teardown to make tests repeatable (like create a new database from scratch at the beginning of each test run, which is what rails tests do)